controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -615,12 +561,12 @@ public void refresh() {
 
     // region private helpers
     @VisibleForTesting
-    String getActiveTxPath(final long epoch, final String txId) {
-        return ZKPaths.makePath(ZKPaths.makePath(activeTxRoot, Long.toString(epoch)), txId);
+    String getActiveTxPath(final int epoch, final String txId) {
```

Comment:
I'm not necessarily saying we need more than an int worth of epochs, but I'm a bit suspicious about downgrading it to int from long. What are we doing it?

Reply:
well we can make it a long without much problem.. but Int.Max is a lot. if a new epoch is generated every second, Int.max (2,147,483,647) will need 68 years before we exhaust the int space.